### ISSUE TITLE
Alire doesn't remove with clause after removing a dependency

### ISSUE THREAD

**Issue by Joebeazelman** [alire-project/alire#1232]
When I execute `alr with aunit --del` it claims `aunit.gpr ` is not found.  The **config/project_config.gpr** file still has  **aunit.gp**r in the declaration. It appears Alire doesn't remove the declaration from the project file.   I was able to reproduce this by:

```
alr with aunit
alr build
alr with aunit --del
alr build -- error messages
```
My project is a library so and Aunit doesn't support libraries according to the build errors. Could build errors prevent the removal of a dependency?

**Comment by Fabien-Chouteau** [comment]
Hi @Joebeazelman 

Does it work if you do a `alr update` before the last build?
```
alr with aunit
alr build
alr with aunit --del
alr update
alr build
```

**Comment by Joebeazelman** [comment]
I tried to remove the crate again and it didn't remove the using clause. When I performed `alr update` it removed it, but with the following messages:

```
warn: There are missing dependencies (use `alr with --solve` for details).
Nothing to update.
warn: Generating possibly incomplete environment because of missing dependencies
warn: Generating possibly incomplete configuration because of missing dependencies
 ```

I performed `alr with --solve`:

/Users/joebeazelman/Projects/Repos/generic_mcu/config/generic_mcu_config.gpr
```
alr with --solve
Dependencies (direct):
   alr*
   aunit^22.0.0
   build*
   gnat_arm_elf^12.1.2
   hal~0.3.0
Dependencies (solution):
   aunit=22.0.0 (origin: source_archive)
   gnat_arm_elf=12.1.2 (installed) (origin: binary_archive)
   hal=0.3.0 (origin: git)
Dependencies (external):
   alr* (direct,missed)
   build* (direct,missed)
Dependencies (graph):
   generic_mcu=0.1.0-dev --> alr*                         
   generic_mcu=0.1.0-dev --> aunit=22.0.0 (^22.0.0)       
   generic_mcu=0.1.0-dev --> build*                       
   generic_mcu=0.1.0-dev --> gnat_arm_elf=12.1.2 (^12.1.2)
   generic_mcu=0.1.0-dev --> hal=0.3.0 (~0.3.0)           
joebeazelman@joebeazelmans-MacBook-Pro generic_mcu % alr with --solve
Dependencies (direct):
   alr*
   aunit^22.0.0
   build*
   gnat_arm_elf^12.1.2
   hal~0.3.0
Dependencies (solution):
   aunit=22.0.0 (origin: source_archive)
   gnat_arm_elf=12.1.2 (installed) (origin: binary_archive)
   hal=0.3.0 (origin: git)
Dependencies (external):
   alr* (direct,missed)
   build* (direct,missed)
Dependencies (graph):
   generic_mcu=0.1.0-dev --> alr*                         
   generic_mcu =0.1.0-dev --> aunit=22.0.0 (^22.0.0)       
   generic_mcu =0.1.0-dev --> build*                       
   generic_mcu =0.1.0-dev --> gnat_arm_elf=12.1.2 (^12.1.2)
   generic_mcu =0.1.0-dev --> hal=0.3.0 (~0.3.0)
   ```
   
I think the "alr, build" dependencies are probably typos when I tried to use `alr with`.  I removed them and things appeared to work, but then I started getting a strange error:

```
alr -d with hal --del
stderr: ADA.IO_EXCEPTIONS.USE_ERROR
stderr: copy of "/Users/joebeazelman/Projects/Repos/generic_mcu/config/alr-enlg.tmp" failed
stderr: raised ADA.IO_EXCEPTIONS.USE_ERROR : copy of "/Users/joebeazelman/Projects/Repos/generic_mcu/config/alr-enlg.tmp" failed
Load address: 0x10243d000
Call stack traceback locations:
0x102ed1cb8 0x102e6f6a7 0x102d48556 0x102920693 0x102b7c206 0x1024faeb2 0x1024fea55 0x10253b59c 0x1025468f8 0x102441b45 0x102444e54

error: copy of "/Users/joebeazelman/Projects/Repos/generic_mcu/config/alr-enlg.tmp" failed
error: alr encountered an unexpected error, re-run with -d for details.
```

**Comment by mosteo** [comment]
That's going to be hard to diagnose without the symbolic traceback :/

I can't reproduce locally, as I was able to add/remove `hal` without encountering that problem. (Although I had to unexpectedly run `alr update` after adding/removing `hal`.)

As a wild guess: may it be caused by a mix of having run `alr` as/not as administrator? I would suggest to try from scratch in a new crate and see if it works then.
